crates/wast/src/binary.rs

```diff
@@ -181,14 +186,34 @@ impl Encode for i32 {
     }
 }
 
-impl Encode for Type<'_> {
+impl Encode for FunctionType<'_> {
     fn encode(&self, e: &mut Vec<u8>) {
         e.push(0x60);
-        self.func.params.len().encode(e);
-        for (_, _, ty) in self.func.params.iter() {
+        self.params.len().encode(e);
+        for (_, _, ty) in self.params.iter() {
             ty.encode(e);
         }
-        self.func.results.encode(e);
+        self.results.encode(e);
+    }
+}
+
+impl Encode for StructType<'_> {
+    fn encode(&self, e: &mut Vec<u8>) {
+        e.push(0x50);
+        self.fields.len().encode(e);
+        for field in self.fields.iter() {
+            (field.mutable as i32).encode(e);
+            field.ty.encode(e);
+        }
+    }
+}
+
+impl Encode for Type<'_> {
+    fn encode(&self, e: &mut Vec<u8>) {
+        match &self.def {
+            TypeDef::Func(func) => func.encode(e),
```

Comment:
Each of these enodings start with a prefix byte (0x60 for functions, 0x50 for structs), would it be possible to move that prefix byte into this `Encode for Type<'_>`? That'd match the style of most other encode-the-enum constructs here too

Reply:
Yup, done.